dogfooding/lib/core/repos/user_auth_repository.dart

```diff
@@ -16,15 +16,21 @@ class UserAuthRepository {
 
   Future<UserCredentials> login() async {
     final response = await videoClient.connect();
-    final userToken = response.getDataOrNull();
-    if (userToken == null) {
-      throw Exception('Failed to connect user');
-    }
-
-    return UserCredentials(
-      token: userToken,
-      userInfo: currentUser,
-    );
+
+    return response.fold(success: (success) {
+      final userToken = response.getDataOrNull();
+
+      if (userToken == null) {
+        throw Exception('Failed to connect user');
+      }
```

Comment:
This isn't going to be null in case of success right?

Reply:
true fixed